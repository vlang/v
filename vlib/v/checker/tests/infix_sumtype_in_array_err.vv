struct OtherType {
	foo string
	bar string
}

type RuneAlias = rune
type RuneAliasOrOtherType = OtherType | RuneAlias

const whitespace = [rune(0x0009), 0x000a, 0x000c, 0x000d, 0x0020]

fn main() {
	mut x := RuneAliasOrOtherType(RuneAlias(rune(`!`)))
	match x {
		RuneAlias {
			if x in whitespace {
				// doing `if x as RuneAlias in whitespace` here
				// works but it should be doing that automatically
				// since I'm inside the RuneAlias match condition.
			}
		}
		else {
			// do something else
		}
	}
}
