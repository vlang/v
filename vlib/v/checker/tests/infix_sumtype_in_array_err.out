vlib/v/checker/tests/infix_sumtype_in_array_err.vv:15:7: error: left operand to `in` does not match the array element type: expected `rune`, not `RuneAliasOrOtherType`
   13 |     match x {
   14 |         RuneAlias {
   15 |             if x in whitespace {
      |                ~~~~~~~~~~~~~~~
   16 |                 // doing `if x as RuneAlias in whitespace` here
   17 |                 // works but it should be doing that automatically
